frontend/src/metabase/common/components/QueryColumnPicker/QueryColumnPicker.tsx

```diff
@@ -48,26 +59,42 @@ export function QueryColumnPicker({
   color = "brand",
   checkIsColumnSelected,
   onSelect,
+  onSelectSegment,
   onClose,
 }: QueryColumnPickerProps) {
-  const sections: Sections[] = useMemo(
-    () =>
-      columnGroups.map(group => {
+  const sections: Sections[] = useMemo(() => {
+    const columnGroupSections: Sections[] = columnGroups.map(
+      (group, groupIndex) => {
         const groupInfo = Lib.displayInfo(query, stageIndex, group);
 
-        const items = Lib.getColumnsFromColumnGroup(group).map(column => ({
-          ...Lib.displayInfo(query, stageIndex, column),
-          column,
-        }));
+        const columnItems = Lib.getColumnsFromColumnGroup(group).map(
+          column => ({
+            ...Lib.displayInfo(query, stageIndex, column),
+            column,
+          }),
+        );
+
+        const includeSegments = !!onSelectSegment && !groupInfo.isFromJoin;
```

Comment:
I [posted in slack](https://metaboat.slack.com/archives/C04CYTEL9N2/p1697148236490379) about this, but currently both MLv1 and MLv2 only support segments for non-joined tables. This seems like a bug to me, but maybe not one worth fixing right now.